assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -0,0 +1,191 @@
+  onError: (error) => { dispatch(paymentFailure(error)); },
+});
+
+// ----- Functions ----- //
+
+const getAmount = (formElements: Object) =>
+  parseFloat(formElements.contributionAmount.value === 'other'
+    ? formElements.contributionOther.value
+    : formElements.contributionAmount.value);
+
+// ----- Event handlers ----- //
+
+let stripeReady = null;
+
+const onSubmit = (e) => {
+  e.preventDefault();
+
+  const { elements } = e.target;
+  const amount = getAmount(elements);
+  const email = elements.namedItem('contributionEmail').value;
+
+  if (stripeReady) {
+    stripeReady.then(() => {
+      openDialogBox(amount, email);
+    });
+  }
+};
+
+
+// ----- Render ----- //
+
+function ContributionForm(props: PropTypes) {
+  const {
+    abParticipations,
+    countryGroupId,
+    selectedCountryGroupDetails,
+    referrerAcquisitionData,
+    optimizeExperiments,
+    currency,
+  } = props;
+
+  let formElement = null;
+
+  const thankYouRoute = `/${countryGroups[countryGroupId].supportInternationalisationId}/thankyou.new`;
+
+  const callback = createTokenCallback({
+    abParticipations,
+    currencyId: currency,
+    referrerAcquisitionData,
+    optimizeExperiments,
+    getData: () => { // eslint-disable-line consistent-return
+      if (formElement) {
+        const elements: any = formElement.elements; // eslint-disable-line prefer-destructuring
+        const firstName = elements.contributionFirstName.value;
+        const lastName = elements.contributionLastName.value;
+        const email = elements.contributionEmail.value;
+        const amount = getAmount(elements);
+
+        return {
+          user: {
+            fullName: `${firstName} ${lastName}`,
+            email,
+          },
+          amount,
+        };
+      }
+    },
+    onSuccess: () => {
+      trackConversion(abParticipations, thankYouRoute);
+      props.onSuccess();
+    },
+    onError: props.onError,
+  });
+
+  stripeReady = setupStripeCheckout(callback, null, currency, props.isTestUser);
+
+  return props.done ?
+    <Redirect to={thankYouRoute} />
+    : (
+      <div className="gu-content__content">
+        <h1>{countryGroupSpecificDetails[countryGroupId].headerCopy}</h1>
+        <p className="blurb">{countryGroupSpecificDetails[countryGroupId].contributeCopy}</p>
+        <ErrorMessage message={props.error} />
+        <form ref={(el) => { if (el) { formElement = el; } }} className={classNameWithModifiers('form', ['contribution'])} onSubmit={onSubmit}>
+          <NewContributionType />
+          <NewContributionAmount
+            countryGroupId={countryGroupId}
+            countryGroupDetails={selectedCountryGroupDetails}
+            currency={currency}
+          />
+          <NewContributionTextInput 
+            id="contributionFirstName" 
+            name="contribution-fname" 
+            label="First Name" 
+            value={window.guardian.user && window.guardian.user.firstName ? window.guardian.user.firstName : null}
```

Comment:
I was told this was the existing behaviour, although I did not find anything such in the one-off logic.

Reply:
It is in here: https://github.com/guardian/support-frontend/blob/master/assets/pages/oneoff-contributions/components/formFields.jsx

```<TextInput
        id="name"
        placeholder="Full name"
        labelText="Full name"
        value={props.fullName.value}
        onChange={props.setFullName}
        onBlur={onFormFieldBlur(props.setFullNameShouldValidate, stripeButtonClassName)}
        modifierClasses={['name']}
        showError={shouldShowError(props.fullName)}
        errorMessage="Please enter your name."
      />```

Basically the props.fullName is initialised in the user state as `""`, and then if there is a user in the window, the user reducer overwrites `""` with the users name

Reply:
Infact, looking at the file, I think rather than reading from the window, we should be passing in these values as props to the component